pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,188 @@
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) factory.Controller {
+
+	ret := &LatencyProfileController{
+		operatorClient:   operatorClient,
+		configClient:     configClient,
+		configMapClient:  v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		configNodeLister: nodeInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		// this is for our general configuration input and our status output in case another actor changes it
+		operatorClient.Informer(),
+
+		// We use nodeInformer for observing current worker latency profile
+		nodeInformer.Informer(),
+
+		// for configmaps of operator client target namespace
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+	).ResyncEvery(time.Minute).WithSync(ret.sync).WithSyncDegradedOnError(operatorClient).ToController(
+		"LatencyProfileController",
+		eventRecorder.WithComponentSuffix("latency-profile-controller"),
+	)
+}
+
+func (c *LatencyProfileController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	_, operatorStatus, _, err := c.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return err
+	}
+
+	// Collect the unique set of revisions of the apiserver nodes
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// Collect the current latency profile
+	desiredApiServerArgumentVals := map[string]string{}
+	configNodeObj, err := c.configNodeLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	switch configNodeObj.Spec.WorkerLatencyProfile {
+	case apiconfigv1.DefaultUpdateDefaultReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.DefaultNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.DefaultUnreachableTolerationSeconds)
+	case apiconfigv1.MediumUpdateAverageReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.MediumNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.MediumUnreachableTolerationSeconds)
+	case apiconfigv1.LowUpdateSlowReaction:
+		desiredApiServerArgumentVals[DefaultNotReadyTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.LowNotReadyTolerationSeconds)
+		desiredApiServerArgumentVals[DefaultUnreachableTolerationSecondsArgument] = strconv.Itoa(apiconfigv1.LowUnreachableTolerationSeconds)
+	}
+
+	condition := metav1.Condition{
+		Type:   apiconfigv1.KubeControllerManagerDegraded,
+		Status: metav1.ConditionUnknown,
+	}
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
```

Comment:
this is a case we should handle right after reading the `c.configNodeLister.Get("cluster")`

Reply:
Updated!